framework/SupervisedLearning/SyntheticHistory.py

```diff
@@ -0,0 +1,202 @@
+# Copyright 2017 Battelle Energy Alliance, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+  Created on Jan 5, 2020
+
+  @author: talbpaul
+  Originally from ARMA.py, split for modularity
+  Uses TimeSeriesAnalysis (TSA) algorithms to train then generate synthetic histories
+"""
+import numpy as np
+
+from utils import InputData, xmlUtils
+import TSA
+
+from .SupervisedLearning import supervisedLearning
+
+class SyntheticHistory(supervisedLearning):
+  r"""
```

Comment:
Does this need to be cast as a raw-string? Also, should we use imperative language in documentation strings? Not to be pedantic, it's your call. If this is our chance to restructure ARMA, I think imperative language keeps things concise and clear, in general.

Reply:
We don't have an imperative tone requirement for docstrings in RAVEN; we're lucky if we get English.

I can remove the raw string casting, though; it's residual from copying from another class.